extensions/layout/init.lua

```diff
@@ -159,13 +159,18 @@ function layout.apply(layout, windowTitleComparator)
                 display = _row[3]()
             elseif fnutils.contains(screen.allScreens(), _row[3]) then
                 display = _row[3]
+            else
+                -- Default to the main screen if the passed-in screen isn't found; useful for
+                -- layouts activated using the screen watcher, meaning that screens in layouts may
+                -- not be in the current screen configuration.
+                display = screen.allScreens()[1]
```

Comment:
Could this use `screen.primaryScreen()` please? Right now it does the same thing as your code, but that may not always be the case.